Foundation/FileManager+Win32.swift

```diff
@@ -39,19 +39,19 @@ extension FileManager {
 
             var wszPathNames: [WCHAR] = Array<WCHAR>(repeating: 0, count: Int(dwCChReturnLength + 1))
             if !GetVolumePathNamesForVolumeNameW(&wszVolumeName, &wszPathNames, DWORD(wszPathNames.count), &dwCChReturnLength) {
-                // TODO(compnerd) handle error
-                continue
+                return nil
             }
 
-            var pPath: DWORD = 0
-            repeat {
-                let path: String = String(decodingCString: &wszPathNames[Int(pPath)], as: UTF16.self)
-                if path.length == 0 {
-                    break
-                }
-                urls.append(URL(fileURLWithPath: path, isDirectory: true))
-                pPath += DWORD(path.length + 1)
-            } while pPath < dwCChReturnLength
+            var pPath: Int = 0
+            while wszPathNames[pPath] != 0 {
+                guard let (url, pathLength): (URL, Int) = wszPathNames.withUnsafeBufferPointer({
```

Comment:
This is confusing, can we not move the base address computation above the rest of the logic rather than re-computing the offset each time?

Reply:
Cleared up the logic a bit so it's easier to read. The base address is now computed by the ArraySlice.